Issue 4112 - Added a CI test

aadhikar opened the pull request:
Bug Description: If the dbhome directory is set, eg to /dev/shm/instance
then an online backup fails because it looks for the log.000000x file
in the wring diretcory.

Relates: https://github.com/389ds/389-ds-base/issues/4112

Reviewed by: ??

aadhikar committed:
Issue 4112 - Added a CI test

Bug Description: If the dbhome directory is set, eg to /dev/shm/instance
then an online backup fails because it looks for the log.000000x file
in the wring diretcory.

Relates: https://github.com/389ds/389-ds-base/issues/4112

Reviewed by: ??

Firstyear commented on dirsrvtests/tests/suites/backups/backup_test.py:
```diff
@@ -64,7 +67,34 @@ def test_missing_backend(topo):
+    bdb_ldbmconfig = BDB_LDBMConfig(topo.standalone)
+    dseldif = DSEldif(topo.standalone)
+    topo.standalone.stop(timeout=10)
+    dseldif.replace(bdb_ldbmconfig.dn, 'nsslapd-db-home-directory', '/tmp/test')
```
It's not a good idea to hardcode in directory names like this because /tmp/test is something I even use and dump values into. You should look at a mktemp or other way of generating randomised folder names.

Firstyear commented on dirsrvtests/tests/suites/backups/backup_test.py:
```diff
@@ -64,7 +67,34 @@ def test_missing_backend(topo):
+    dseldif = DSEldif(topo.standalone)
+    topo.standalone.stop(timeout=10)
+    dseldif.replace(bdb_ldbmconfig.dn, 'nsslapd-db-home-directory', '/tmp/test')
+    topo.standalone.start(timeout=10)
```
Don't hardcode timeout in start/stop because on a highly loaded CI system, 10 seconds may not be enough. There is a default value here for a reason. Please remove the timeout.

aadhikar committed:
Removed hard coded temp directory and timeout in instance start/stop.

droideck reviewed (approved):
Looks good to me!

Firstyear commented on dirsrvtests/tests/suites/backups/backup_test.py:
```diff
@@ -64,7 +68,37 @@ def test_missing_backend(topo):
+    bdb_ldbmconfig = BDB_LDBMConfig(topo.standalone)
+    dseldif = DSEldif(topo.standalone)
+    topo.standalone.stop()
+    cmd = ['mktemp', '-d']
```
https://docs.python.org/3/library/tempfile.html

you don't need subprocess. Whenever possible, subprocess is a last resort, always look for a python api first.

aadhikar committed:
Issue 4112 - Added a CI test

Bug Description: If the dbhome directory is set, eg to /dev/shm/instance
then an online backup fails because it looks for the log.000000x file
in the wring directory.

Relates: https://github.com/389ds/389-ds-base/issues/4112

Reviewed by: Firstyear,droideck (Thanks!)
